File: ProiectPOO/CommandPanel.cpp
#include<iostream>
#include"CommandPanel.h"
#include"RecipeCake.h"
#include"CommandTaker.h"
#include"Cake.h"

using namespace std;

CommandPanel::CommandPanel() {}

void CommandPanel::showProducts() {
	for (RecipeCake product : menu) {
		cout << "Numele produsului: " << product.getName() << endl;
		cout << "Timpul de preparare al produsului (in secunde): " << product.getTime() << endl;
	}
}

void CommandPanel::selectProduct(string name) {
	Cake product;
	int search = 0;
	for (RecipeCake recipe : menu) { //se cauta numele prajiturii in meniu
		if (recipe.getName() == name) {
			search = 1;
			commandTaker.carouselRecipe = recipe;
			product = commandTaker.takeCommand(recipe); //daca se gaseste se plaseaza comanda
			cout << "Prajitura a fost servita !" << endl;
		}
	}
	if (search == 0)
		cout << "Produsul nu exista in meniu !" << endl;
}

void CommandPanel::selectProduct(string name, int numberOfProducts) {
	Cake* products;
	int search = 0;
	for (RecipeCake recipe : menu) { //se cauta numele prajiturilor in meniu
		if (recipe.getName() == name) {
			search = 1;
			commandTaker.carouselRecipe = recipe;
			products = commandTaker.takeCommand(recipe, numberOfProducts); //daca se gasesc se plaseaza comanda
			cout << "Prajiturile au fost servite !" << endl;
		}
	}
	if (search == 0)
		cout << "Produsele nu exista in meniu !" << endl;
}

void CommandPanel::showProductsInCarousel() {
	Cake* carouselStorage;
	Cake products[12];
	carouselStorage = commandTaker.getCakesFromCarousel(); //aici se afla prajiturile din depozit
	for (int i = 0; i < 12; i++) {
		products[i] = *(carouselStorage + i);
	}
	cout << "Produsele din carusel sunt : " << endl;
	for (int i = 0; i < 12; i++) {
		if (products[i].getName() != "empty") //se cauta pozitiile ocupate de prajituri
			cout << products[i].getName() << " "; //si se afiseaza numele lor
	}
	cout << endl;
}

File: CakeMakerProj/CommandPanel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CakeMakerProj
{
    public class CommandPanel
    {
        private List<RecipeCake> Menu;
        private ICommandTaker CommandTaker;

        public CommandPanel()
        {
            this.Menu=new List<RecipeCake>();
            this.CommandTaker=new CommandTaker();
        }

        public void SetMenu(List<RecipeCake> menu)
        {
            this.Menu = menu; 
        }

        public void SetCommandTaker(ICommandTaker commandTaker)
        {
            this.CommandTaker = commandTaker;
        }

        public void ShowProducts()
        {
            foreach (RecipeCake product in Menu)
            {
                Console.WriteLine("Product Name: " + product.GetName());
                Console.WriteLine("Preparation Time (in seconds): " + product.GetTime());
            }
        }

        public void SelectProduct(string name)
        {
            Cake product;
            int search = 0;
            foreach (RecipeCake recipe in Menu)
            {
                if (recipe.GetName() == name)
                {
                    search = 1;
                    CommandTaker.SetRecipeCake(recipe);
                    product = CommandTaker.TakeCommand(recipe);
                    Console.WriteLine("The cake has been served!");
                }
            }

            if (search == 0)
                Console.WriteLine("The product does not exist in the menu!");
        }

        public void SelectProduct(string name, int numberOfProducts)
        {
            Cake[] products;
            int search = 0;
            foreach (RecipeCake recipe in Menu)
            {
                if (recipe.GetName() == name)
                {
                    search = 1;
                    CommandTaker.SetRecipeCake(recipe);
                    products = CommandTaker.TakeCommand(recipe, numberOfProducts);
                    Console.WriteLine("The cakes have been served!");
                }
            }

            if (search == 0)
                Console.WriteLine("The products do not exist in the menu!");
        }

        public void ShowProductsInCarousel()
        {
            Cake[] carouselStorage;
            Cake[] products = new Cake[12];
            carouselStorage = CommandTaker.GetCakesFromCarousel();

            for (int i = 0; i < 12; i++)
            {
                products[i] = carouselStorage[i];
            }

            Console.WriteLine("The products in the carousel are:");
            for (int i = 0; i < 12; i++)
            {
                if (products[i].GetName() != "empty")
                    Console.Write(products[i].GetName() + " | ");
            }
            Console.WriteLine();
        }
    }
}
